### PR TITLE
python: add support to multiple python versions(27, 35, 36, dev)

### PR DESCRIPTION
**Pull request by lucasts**
how it works:
It will respect `.python-version` if it exists and use a valid version.
It will not install other versions, but will allow devs to select a stable version from main python series(2.7, 3.5, 3.6) offered by our python platforms.

PR todo: 
- [x]  multiple python series
- [x] tests running
- [x] real live tests deploying apps
- [x] update readme

future nice to haves:
 - allow direct series selection instead of a full version (eg: `3.6` or `3.6.2`) 
 - install more python versions by default( latest + latest - 1 and so fort) to allow apps to migrate with easy
 - pypy and possible other python runtimes.

### CODE REVIEW

File: python/deploy [hunk 1]
```diff
@@ -8,13 +8,31 @@ SOURCE_DIR=/var/lib/tsuru
 source ${SOURCE_DIR}/base/deploy
 source ${SOURCE_DIR}/base/rc/config
 
-APP_VENV=/home/application/.app_env
+APP_VENV=app_env
 
+#find apps python version
+PYTHON_VERSION=3.5.3  # default version if none pointed
```
Comment (andrestc) [inline python/deploy]: Maybe we could check if PYTHON_VERSION env is already set? So one can extend our public platform and change the default python version with easy. We do that for the ruby platform: https://github.com/tsuru/platforms/blob/master/ruby/deploy#L29

File: python/install [hunk 2]
```diff
@@ -9,12 +9,29 @@ APP_VENV=/home/application/.app_env
...
+export CFLAGS='-O2'
+
+for pyversion in $PY_VERSIONS; do (pyenv install $pyversion); done
 
-virtualenv $APP_VENV
-$APP_VENV/bin/pip install --upgrade pip distribute
-echo "export PATH=${APP_VENV}/bin:\${PATH}" | tee -a ${HOME}/.profile /etc/profile >/dev/null
-echo "Defaults secure_path=\"${APP_VENV}/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin\"" > /etc/sudoers.d/venv
-chown -R ${USER}:${USER} ${APP_VENV}
+chown -R ubuntu:ubuntu $PYENV_ROOT
```
Comment (lucasts) [inline python/install]: currently improving path exporting and chown

File: python/README.md [hunk 3]
```diff
@@ -1,6 +1,22 @@
...
+ex:
+```
+3.6.1
+```
+
+available python versions:
+- 2.7.13
+- 3.5.3
+- 3.6.1
+- 3.7-dev
```
Comment (lucasts) [inline python/README.md]: make sense to have so many series?

heroku is enable only 2.7.13 and 3.6.1.

File: python/deploy (hunk 1 above)
Comment (lucasts) [inline python/deploy]: done ;)

File: python/README.md (hunk 3 above)
Comment (lucasts) [inline python/README.md]: done

**Comment by cezarsa** [comment]
This is great, thanks!

**Comment by cezarsa** [comment]
I made a few changes to the virtualenv naming to ensure that it's recreated if the python version changes in https://github.com/tsuru/platforms/commit/e34442b454381f2313fb43ec999a1bcddc7d215b. Can you take a look to see if this can be an issue?

**Comment by lucasts** [comment]
not at all @cezarsa. I honestly just forgot about it.
It would be nice to test this scenario tought. 
I will try to write something this week.

File: python/deploy [hunk 4]
```diff
@@ -8,13 +8,31 @@ SOURCE_DIR=/var/lib/tsuru
...
+
+# Create a app env if needed (with correct python version)
+if [ ! -e $PYENV_ROOT/versions/$APP_VENV ]; then
+    pyenv virtualenv ${PYTHON_VERSION} ${APP_VENV}
+fi
+
+# install projects deps with pip
 pushd $CURRENT_DIR
+pyenv global $APP_VENV
+
```
Comment (lucasts) [inline python/deploy]: nice catch, seems fair.

File: python/install [hunk 5]
```diff
@@ -9,12 +9,29 @@ APP_VENV=/home/application/.app_env
 SOURCE_DIR=/var/lib/tsuru
 source ${SOURCE_DIR}/base/rc/config
 
-add_repository ppa:fkrull/deadsnakes-python2.7
+PY_VERSIONS="2.7.13 3.5.3 3.6.1 3.7-dev"
+PYENV_ROOT="/var/lib/pyenv"
+
 apt-get update
-apt-get install python-virtualenv python-pip python-dev git -y
+apt-get install -y make build-essential libssl-dev zlib1g-dev libbz2-dev \
```
Comment (lucasts) [inline python/install]: there is another commit @cezarsa did in the master branch tooking care of that.

Comment (lucasts) [inline python/install]: here: https://github.com/tsuru/platforms/commit/a8e5f8f17f03a931efbaf4da2721fa373c6a2bf1

**Pull request merged**